x-pack/plugins/licensing/common/license_update.ts

```diff
@@ -29,7 +30,7 @@ export function createLicenseUpdate(
 ) {
   const manuallyRefresh$ = new Subject<void>();
 
-  const fetched$ = merge(triggerRefresh$, manuallyRefresh$).pipe(
+  const fetched$ = merge(triggerRefresh$, manuallyRefresh$.pipe(debounceTime(1000))).pipe(
```

Comment:
Some plugins call `licensing.refresh()` (exposed on the licensing plugin contract), and they are awaiting for the licence back.

Using `debounceTime` here might have a negative impact, as it will add an extra 1 second delay systematically to those calls. `throttleTime` is probably a better alternative, but I have to confirm it works, as the description says subsequent emissions are "ignored".

Reply:
Yeah, good catch, but actually both are problematic. Using `throttleTime` will ignore the subsequent emissions, meaning that the wait for `refresh` could be from up to 30s (the interval based refresh)

TBH, I only added this because I thought it would be a quick win. As we've seen, the use of `exhaustMap` is likely sufficient.

So probably the correct thing to do is to revert this change and keep the code as it was, wdyt?

Reply:
**UPDATE**: `throttleTime` has a 3rd parameter, configuration, that serves precisely this purpose:
In this case: `{ leading: true, trailing: true }` is what we need for our use case.